src/lib/storage/frame_of_reference_segment/frame_of_reference_encoder.hpp

```diff
@@ -106,6 +109,10 @@ class FrameOfReferenceEncoder : public SegmentEncoder<FrameOfReferenceEncoder> {
 
     auto compressed_offset_values = compress_vector(offset_values, vector_compression_type(), allocator, {max_offset});
 
+    if (!segment_contains_null_values) {
```

Comment:
Minimal code style thing, again not a must have: You could also switch the two cases and get rid of the negation. Personally, I would consider the two cases to be on eye-level, so this wouldn't be a guard clause type of thing for me.

Reply:
Nice catch. Changed.